wgetpaste

```diff
@@ -193,9 +193,14 @@ tcl tcsh tea tex text text treetop ts urbiscript vala vb.net velocity verilog vg
 xml+cheetah xml+django xml+evoque xml+lasso xml+mako xml+myghty xml+php xml+erb xml+smarty \
 xml+velocity xml xquery xslt xtend yaml"
 EXPIRATIONS_pinnwand="1day 1week 1month never"
-POST_pinnwand="submit=Paste! % % lexer expiry % code"
-REGEX_LOC_pinnwand="\(/show/[^ ]*\).*$|https://bpaste.net\1"
-REGEX_RAW_pinnwand='s|^\(https\?://[^/]*/\)show\(/[^ ]*/\?\)$|\1raw\2|'
+REGEX_URL_pinnwand='s|^.*"link": "\([^"]\+bpa.st[^"]\+\)".*$|\1|p'
+REGEX_RAW_pinnwand='s|^\(https://bpa.st\)\(/.*\)$|\1/raw\2|'
+escape_description_pinnwand() { sed -e 's|"|\\"|g' -e 's|\x1b|\\u001b|g' -e 's|\r||g' <<< "$*"; }
+escape_input_pinnwand() { sed -e 's|\\|\\\\|g' -e 's|\x1b|\\u001b|g' -e 's|\r||g' -e 's|\t|\\t|g' -e 's|"|\\"|g' -e 's|$|\\n|' <<< "$*" | tr -d '\n'; }
+json_pinnwand() {
+    local description="${1}" language="${2}" content="${3}" expiration="${4}"
+	echo "{\"expiry\": \"${expiration}\", \"files\": [ {\"name\": \"${description}\", \"lexer\": \"${language}\", \"content\": \"${content}\"} ] }"
```

Comment:
I recommend doing the following:

```suggestion
	jq -Mnc ".expiry = \"${expiration}\" | .files[0] = { name: \"${description}\", lexer: \"${language}\", content: \"${content}\" }"
```

It will add a dependency on the `jq` utility; however, `jq` can also be used in `geturl` to parse JSON output instead of relying on regular expressions.

Using `jq` also enables the removal of some of the escape functions used below.

Reply:
Maybe we should discuss this on another issue, because that change would impact all other engines as well.